### DISCUSSION TITLE
Using with React Native

### DISCUSSION THREAD

**Discussion by k6jge3jf10o7** [pmndrs/react-three-rapier#375] [Q&A]
Is it possible to use react-three-rapier with react-three-fiber in React Native project? Is there any example how to do this?

**Comment by wiledal** [comment]
I haven't seen this usecase, but seeing as it is just javascript, I believe it could work.

**Comment by imvikysharma** [comment]
I suppose it is not possible to use react-three-rapier with react-three-fiber in a React Native project as react-three-fiber is a library for rendering 3D graphics using WebGL, which is not available on React Native. React Native uses different rendering engines for iOS and Android, neither of which are based on WebGL.
However, there are alternatives for rendering 3D graphics in React Native, such as react-native-3d-model-view, react-native-gl-model-view,

I would like to recommend a book on React, I am using it personally and I highly recommend it.

https://www.amazon.com/dp/B0BVRKSY6G?maas=maas_adg_A78FB1C22F0F576564CF29D2524311F0_afap_abs&ref_=aa_maas&tag=maas

**Reply by wiledal** [reply]
@imvikysharma React-Three-Fiber has native bindings in @react-three/fiber/native. It uses expo-gl for the native bindings.
https://docs.pmnd.rs/react-three-fiber/getting-started/installation#react-native

On second thought, this whole response sounds oddly like a ChatGPT response, and it's promoting an unrelated book, coming from an account with zero commits, seemingly unaware of the context of the discussion. I'm 90% convinced this is a bot.

**Reply by imvikysharma** [reply]
I am not a bot bro..... I am new here

**Reply by wiledal** [reply]
That's exactly what a bot would say! Jokes aside, the vague response with repeated context keywords ending with an affiliate product link was very bot-like. People without React experience don't usually find themselves on this niche corner of github 😄 

I apologise for the confusion. I recommend you read the docs for `react-three-fiber` (React Native usage is mentioned on the first page), and I hope I didn't put you off from commenting further!

**Comment by codingnuclei** [comment]
hey 👋 

As mentioned before r3f has support for react native... has anyone tried it with r3rapier, an built anything complex?
